### DISCUSSION TITLE
setScrollPosition is not a function

### DISCUSSION THREAD

**Discussion by bagladivyang03** [quasarframework/quasar#10145] [CLI - PWA mode] [answered]
I m trying to use the scroll to an element and I m trying like this 

```
scrollToElement(id) {
      let el = document.getElementById(id)
      console.log(el)
      const target = getScrollTarget(el)
      const offset = el.offsetTop - 0
      console.log(offset)
      const duration = 500
      setScrollPosition(target, offset, duration)
},
```
            
            
 And I have also imported the scroll 
`  import {scroll} from 'quasar';`
`const {getScrollTarget, setScrollPosition} = scroll`

and my div tag code 
```
    <div id='home'>
    <section  class="xs:tw-pt-20 xl:tw-pt-0" style="min-height: 70vh;">
      <q-card class="tw-bg-blue-100 no-shadow text-center tw-text-gray-800 " style="padding-top: 15%">
        <q-card-section>
          <span class="xs:tw-text-2xl xl:tw-text-5xl">👋 Hi, my name is</span>
          <div class="xs:tw-text-4xl xl:tw-text-7xl q-ma-md tw-font-extrabold text-color1">
            Divyang Bagla
          </div>
          <div class="tw-mt-10 xs:tw-text-xl xl:tw-text-4xl tw-font-bold text-color2">
            I am Final Year CSE Student from MIT-WPU.
          </div>
          <div class="xs:tw-text-lg xl:tw-text-3xl tw-mt-12 tw-font-semibold" style="color: #f4b825;">
            WEB DEVELOPMENT | DEVELOPER | LEARNER
          </div>
        </q-card-section>
      </q-card>
    </section>
    </div>
```

and the code from where I am clicking to scroll to that element.

```
            <q-tab class="q-mr-sm q-py-xs"
                   @click="scrollToElement('home')" style="width:120px;min-height:auto !important;color: white"
                   label="Home"/>
```
And I m getting the below error 
![image](https://user-images.githubusercontent.com/57760543/126449990-9c54d6f5-b453-4b36-acef-9d305018e07f.png)

Anyone can pls see and help me asap :).

**Comment by bagladivyang03** [comment]
@MilosPaunovic @hawkeye64  can u pls help

**Comment by MilosPaunovic** [comment, marked as answer]
Hello @bagladivyang03,

Have you checked [this](https://quasar.dev/start/upgrade-guide#scroll-utils) out?
It's a part of the Upgrade guide from Q1 to Q2 regarding Scroll utils.

**Reply by bagladivyang03** [reply]
Thanks @MilosPaunovic !!
It works

**Comment by pdanpdan** [comment]
Why are you pinging people?
Make a codepen with what you have done, and while doing so I suppose you'll find out you haven't imported the function where you are using it.

**Reply by bagladivyang03** [reply]
import {scroll} from 'quasar';
const {getScrollTarget, setScrollPosition} = scroll

I have imported this

What else should I impose 


and sorry for pinging

**Reply by pdanpdan** [reply]
setVerticalScrollPosition as Milos mentioned

**Reply by bagladivyang03** [reply]
Yeah thanks got it @pdanpdan !!!
